src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -184,6 +186,42 @@ private async Task<IWebHost> BuildWebHost(ScriptApplicationHostOptions hostOptio
             return settings;
         }
 
+        /// <summary>
+        /// Start up authentication middleware process
+        /// </summary>
+        /// <param name="listenUrl">Where clients send requests</param>
+        /// <param name="hostUrl">Where middleware authentication container sends its modified requests. Also where the Functions runtime will be listening to</param>
+        /// <param name="certPath">Full (not relative) path to local certificate (used for HTTPS)</param>
+        /// <param name="certPassword">Password used when creating above certificate</param>
+        /// <returns>Started task</returns>
+        private async Task StartAuthenticationProcessAsync(string listenUrl, string hostUrl, string certPath, string certPassword)
+        {
+            // Listen URL: Where client requests come in to
+            // Host Destination URL: Output location of middleware authentication container. This is what the Functions runtime will be listening to.
+            if (CommandChecker.CommandExists("dotnet"))
+            {
+                // TODO IMPORTANT - this is very much just for dev/testing. NOT shippable.
+                // The Middleware Linux Nuget is not yet available. 
+                // Once it is, this code will be changed to invoke that, rather than a hardcoded DLL
+                string dllPath = @"C:\wagit\AAPT\Antares\EasyAuth\src\EasyAuth\MiddlewareLinux\bin\Debug\netcoreapp2.0\MiddlewareLinux.dll";
+
+                string query = $"--{Constants.MiddlewareListenUrlSetting} {listenUrl} --{Constants.MiddlewareHostUrlSetting} {hostUrl} " +
+                    $"--{Constants.MiddlewareLocalSettingsSetting} {SecretsManager.MiddlewareAuthSettingsFileName}";
+                if (certPath != null && certPassword != null)
+                {
+                    query += $"  --{Constants.MiddlewareCertPathSetting} {certPath} --{Constants.MiddlewareCertPasswordSetting} {certPassword}";
+                }
+                var process = Process.Start("CMD.exe", $"/C dotnet {dllPath} {query}");
+                var cancellationToken = new CancellationTokenSource();
+                cancellationToken.Token.Register(() => process.Kill());
+                await process.WaitForExitAsync();
```

Comment:
I'm a bit unsure about the use of the `cancellationToken` here. It doesn't look like you really need it. You can also use the `Executable` class I have defined to help with this. The usage is fairly straight forward. Check [here ](https://github.com/Azure/azure-functions-core-tools/blob/299805e0edad5cb895c5f0d3d432ad847d8d721a/src/Azure.Functions.Cli/Helpers/DotnetHelpers.cs#L139)  or [here](https://github.com/Azure/azure-functions-core-tools/blob/299805e0edad5cb895c5f0d3d432ad847d8d721a/src/Azure.Functions.Cli/Helpers/PythonHelpers.cs#L65) for samples